### PR TITLE
Added divisor_override to avg_pool2d and avg_pool3d for compatibility with pytorch

### PR DESCRIPTION
**Pull request by ifedan**
https://github.com/pytorch/xla/issues/830

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -579,7 +580,8 @@ at::Tensor AtenXlaType::avg_pool2d(const at::Tensor& self,
 at::Tensor AtenXlaType::avg_pool2d_backward(
     const at::Tensor& grad_output, const at::Tensor& self,
     at::IntArrayRef kernel_size, at::IntArrayRef stride,
-    at::IntArrayRef padding, bool ceil_mode, bool count_include_pad) {
+    at::IntArrayRef padding, bool ceil_mode, bool count_include_pad, c10::optional<int64_t> divisor_override) {
+  // devisor_override is set not supported yet.
```
Comment (asuhan) [inline torch_xla/csrc/aten_xla_type.cpp]: Look at the line below, you can add `divisor_override` for compatibility. It's not correct to ignore it.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 2]
```diff
@@ -564,11 +564,14 @@ at::Tensor AtenXlaType::avg_pool2d(const at::Tensor& self,
                                    at::IntArrayRef kernel_size,
                                    at::IntArrayRef stride,
                                    at::IntArrayRef padding, bool ceil_mode,
-                                   bool count_include_pad) {
+                                   bool count_include_pad,
+                                   c10::optional<int64_t> divisor_override) {
+  // devisor_override is set not supported yet.
```
Comment (asuhan) [inline torch_xla/csrc/aten_xla_type.cpp]: `devisor_override` -> `divisor_override`

Comment (dlibenzi) [reply torch_xla/csrc/aten_xla_type.cpp]: I'd just drop this comment as it does not add much one cannot read from the code.
Same in all other instances of this PR.

File: torch_xla/csrc/ops/native_batch_norm_backward.cpp [hunk 3]
```diff
@@ -34,14 +34,13 @@ xla::Shape NodeOutputShape(const Value& grad_out, const Value& input,
...
     const Value& save_mean, const Value& save_invstd, bool training, double eps)
-    : Node(
-          ir::OpKind(at::aten::native_batch_norm_backward),
-          {grad_out, input, weight, save_mean, save_invstd},
-          [&]() {
-            return NodeOutputShape(grad_out, input, weight, save_mean,
-                                   save_invstd, training);
-          },
-          /*num_outputs=*/3, xla::util::MHash(training, eps)),
+    : Node(ir::OpKind(at::aten::native_batch_norm_backward),
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/native_batch_norm_backward.cpp]: We should not be using global reformat scripts as they tend to intermingle real changes with formatting changes.
Please drop those. I will fix them in a dedicated PR.

**Comment by dlibenzi** [comment]
I just merged this and will be fixing issues in another PR, as we are RED otherwise.

**Pull request merged**